plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseDataSource.java

```diff
@@ -0,0 +1,656 @@
+        super.refreshObject(monitor);
+        
+        this.tablespaceCache.clearCache();
+        this.userCache.clearCache();
+        this.roleCache.clearCache();
+        this.replCache.clearCache();
+        
+        hasStatistics = false;
+        
+        this.initialize(monitor);
+
+        return this;
+    }
+    
+    @Nullable
+    @Override
+    public <T> T getAdapter(Class<T> adapter) {
+        if (adapter == DBCServerOutputReader.class) {
+            return adapter.cast(outputReader);
+        } else if (adapter == DBCQueryPlanner.class) {
+            return adapter.cast(new AltibaseQueryPlanner(this));
+        } else if (adapter == DBAServerSessionManager.class) {
+            return adapter.cast(new AltibaseServerSessionManager(this));
+        }
+        
+        return super.getAdapter(adapter);
+    }
+    
+    @Override
+    protected Connection openConnection(
+            @NotNull DBRProgressMonitor monitor, 
+            @Nullable JDBCExecutionContext context, 
+            @NotNull String purpose) throws DBCException {
+        try {
+            Connection connection = super.openConnection(monitor, context, purpose);
+            try {
+                for (SQLWarning warninig = connection.getWarnings();
+                    warninig != null && !isPasswordExpireWarningShown;
+                    warninig = warninig.getNextWarning()
+                ) {
+                    if (checkForPasswordWillExpireWarning(warninig)) {
+                        isPasswordExpireWarningShown = true;
+                    }
+                }
+            } catch (SQLException e) {
+                log.debug("Can't get connection warnings", e);
+            }
+            return connection;
+        } catch (DBCException e) {
+            throw e;
+        }
+    }
+    
+    private boolean checkForPasswordWillExpireWarning(@NotNull SQLWarning warning) {
+        if ((warning != null) && (warning.getErrorCode() == AltibaseConstants.EC_PASSWORD_WILL_EXPIRE)) {
+            DBWorkbench.getPlatformUI().showWarningMessageBox(
+                    AltibaseConstants.SQL_WARNING_TITILE,
+                    warning.getMessage() + 
+                    AltibaseConstants.NEW_LINE + 
+                    AltibaseConstants.PASSWORD_WILL_EXPIRE_WARN_DESCRIPTION);
+            
+            return true;
+        }
+        return false;
+    }
+        
+    @NotNull
+    @Override
+    public AltibaseDataSource getDataSource() {
+        return this;
+    }
+
+    @Override
+    public boolean splitProceduresAndFunctions() {
+        return true;
+    }
+    
+    @NotNull
+    @Override
+    public Class<? extends DBSObject> getPrimaryChildType(@Nullable DBRProgressMonitor monitor) throws DBException {
+        return AltibaseSchema.class;
+    }
+
+    public DBSObject findSynonynTargetObject(DBRProgressMonitor monitor, @Nullable String refSchemaName, @NotNull String refObjName)
```

Comment:
Typo. findSynonymTargetObject

Reply:
Oops...I'll fix it. Anyway, this is a really confusing spelling :)